source/service.go

```diff
@@ -717,4 +717,31 @@ func (sc *serviceSource) AddEventHandler(ctx context.Context, handler func()) {
 	// Right now there is no way to remove event handler from informer, see:
 	// https://github.com/kubernetes/kubernetes/issues/79610
 	sc.serviceInformer.Informer().AddEventHandler(eventHandlerFunc(handler))
+
+	// Check updates in headless endpoints.
+	// It's ignores adds and deletes to avoid unnecessary handler processing.
+	sc.endpointsInformer.Informer().AddEventHandler(headlessEndpointFilter{eventHandlerFunc(handler)})
```

Comment:
I would rather like to see an opt-in approach rather than having this added.

Reply:
so, do I need to do something?

Reply:
@szuecs

Reply:
@szuecs @mloiseleur Can you please reply?

Reply:
I imagine and if / else branch to setup the informer.
opt-in means you have to provide a flag to enable it.

Reply:
🙈 
isn't it strange behavior? headless services can be created, but updating only by some flag?
What about the `watch-headless` flag for all cases, creating/deleting and updating? Is it ok?

Reply:
@szuecs?

Reply:
/lgtm cancel

@tufitko @orgads Thanks for your patience. You're right the flag should support all possible operations, not only updating. There is no `watch-headless` flag FTM in `--help` of external-dns, AFAIK.

Looking at the help, I see two options:
1. Adding a new explicit type in `--service-type-filter`, not enabled by default
2. Adding it in `--source`

In both cases, it should support `--label-filter`, `--namespace` and `--annotation-filter`. FTM, option 1 makes more sense to me.

Wdyt ?

Reply:
If I understand correctly, @tufitko suggested to add this flag.

Reply:
Oh, sorry for the delay. I forgot about this problem. I'll try to implement the first option.

Reply:
@mloiseleur Are you sure it’s not enabled by default? It would be strange behavior if you don’t use the filter but still don’t get everything. Additionally, this filter seems to work differently compared to the others.

Reply:
hmm, I think it better to add explicit type in `--service-type-filter`, enabled by default.
Anyway, it's breaking changes. 

With a new source, 2 sources are triggered when creating/updating/deleting services (headless or not) because there isn't any filter in the event handler. Is that OK?

With a new service type (enabled by default), the source can be triggered more often. 
For users, who use external-dns for headless services, this change would be good (allows increasing sync interval)
For users, who don't have headless services in a cluster, nothing is changed
Bad case, when a user has a headless service, but doesn't use external-dns for managing records

Reply:
I think the last proposal from @tufitko makes sense to me. I think this is more of a bug fix than something we need to carefully feature flag and it's gonna be in the new release only anyway. wdyt @mloiseleur ?

Reply:
@Raffo that works for me. 

@tufitko any questions left ?